components/builder-http-gateway/src/http/middleware.rs

```diff
@@ -246,7 +227,32 @@ impl AfterMiddleware for Cors {
     }
 }
```

Comment:
This entire function is SO MUCH BETTER now than it was before.  Wow.

Reply:
It took a lot of head beating... :)